src/Pages/SignUpPage/SignUpPage.tsx

```diff
@@ -41,28 +54,34 @@ const SignUpPage: React.FC = () => {
                     required
                     label="Password"
                     type="password"
-                    variant="outlined"
+                    variant="filled"
                     margin="dense"
                     value={password}
-                    onChange={(e) => setPassword(e.target.value)}
+                    onChange={handlePasswordChange} // Handle password change
                     className="sign-up-input"
+                    error={passwordError !== ''}
+                    helperText={passwordError}
                 />
-                <FormControlLabel required
-                                  control={
-                                    <Checkbox sx={{color: blue[50]}}
-                                              checked={isCheckboxChecked}
-                                              onChange={() => setIsCheckboxChecked(!isCheckboxChecked)}
-                                    />
-                }
-                                  label={
+                <FormControlLabel
+                    required
+                    control={
+                        <Checkbox
+                            sx={{ color: blue[50] }}
+                            checked={isCheckboxChecked}
+                            onChange={() => setIsCheckboxChecked(!isCheckboxChecked)}
+                        />
+                    }
+                    label={
                                     <Typography variant="body1" style={{ fontSize: '14px' }}>
                                       I accept all terms and conditions
-                                  </Typography>}/>
+                                  </Typography>}
+                />
                                   
-                <Button variant="contained"
-                        className="sign-up-button"
-                        onClick={handleLogin}
-                        disabled={!isCheckboxChecked}
+                <Button
+                    variant="contained"
+                    className="sign-up-button"
+                    onClick={handleLogin}
+                    disabled={!isCheckboxChecked || !isFormFilled()}
```

Comment:
@rooshnie23 The button should also be disabled if the password error is not an empty string

Reply:
Could you elaborate on this? I don't seem to understand.

Reply:
If we see that the passwordError is not empty, which means the password isn't 8 characters yet, then the button should be disabled as well. It should only get enabled once everything is filled out as well as the password having 8 characters.